new/community-slack.md

```diff
@@ -0,0 +1,170 @@
+have been happy with them, but those hitting hard limits on the Slack side have
+had zero recourse but to abort and move elsewhere.
+
+## Alternatives
+
+For the two problem cases mentioned above, they chose to migrate to [Gitter](https://gitter.im/)
+and [Discord](https://discordapp.com/) respectively. Gitter doesn't have the
+same level of mobile device support and might confuse some users as our
+support discussions generally happen in a single, shared channel rather than
+the smattering of repositories that make up Chef, Chef Server, etc. Discord
+seems reasonable as an alternative, but the market penetration of Slack is a
+strong nod in their favor.
+
+There are a few generally-similar FOSS options
+([Mattermost](http://www.mattermost.org/), [Zulip](https://www.zulip.org/), etc)
+but none seem to be well polished enough to be worth migrating to at this point.
+Additionally most require self-hosting, which would increase the operational
+load on Chef Software's staff.
+
+Another option is to simply stick with IRC, and try to set up a mix of hosted
+bouncers/proxies and an improved web client. While this is possible, I've not
+heard of any other major community providing this level of service for IRC.
+
+## Problems
+
+The biggest problem is mentioned above, unknown and nebulous user limits. We
+will not be butting up against those immediately, but based on IRC usage numbers
+I would expect us to be within striking distance within a year. This could
+probably be solved with some level of pruning of idle users, only 14% of our
+yearly IRC users have been active in the last month, but that brings really
+gross UX with it. Additionally our projections are based on IRC, which could be
+a drastic under-estimation as the improved UX of Slack encourages more users to
+participate.
+
+The second big problem is that Slack requires making your email address used
+for the account public. Wordpress has built a workaround for this in the form
+of a `chat.wordpress.org` mailserver that forwards based on your Wordpress.org
+account information and only accepts email from Slack. We could take a similar
+approach using Chef Supermarket logins, but this would require either augmenting
+SlackIn or developing something custom.
+
+The third big problem is the overall new user flow. Slack is designed assuming
+a company structure, so it does not offer any form of public chat or
+user-initiated signup. SlackIn offers this, but the UX is a bit awkward compared
+to normal web application signup.
+
+A fourth general systemic issue is that Slack, as a company, seems to not
+consider this to be a use case worth pursuing. This means they have [stated
+directly](https://www.quora.com/Is-there-a-way-to-ignore-certain-users-in-Slack)
+that they do not plan to offer any community control or moderation features.
+This means no user ignores, no kick/ban abilities, no way to mark trusted users
+(equivalent to op/mod in IRC). Slack also doesn't offer public logs, though they
```

Comment:
Or that we would hook up the existing log bot via Slack's IRC gateway? I for one would be really sad if they went away which is why I listed it as a problem we need solve :)